### ISSUE TITLE
Add per-form permission

### ISSUE THREAD

**Issue by owenmorrill** [CuBoulder/express#1992]
## Context

We need a way to give permission to users to view submissions from each form. So if a form owner wants to share the submissions with someone else, they'd just need to add that user to the webform in some way and that gives that user access.

## Process

Anyone with create/edit permissions to the webform need to be able to allow any user on the site access to their webform's results.

![cursor_and_edit_webform_test___university_of_colorado_boulder](https://user-images.githubusercontent.com/385234/37178599-2b6d5a70-22e0-11e8-8204-45dec3036092.png)

**Comment by owenmorrill** [comment]
From #1869 
"A new user reference field needs to be added to the webform content type and a permission check added when viewing submissions. The permission to view submissions of a specific webform will be managed by the form's owner. Only SO will automatically have access to create forms and view all results. Even other users with the FO will have to granted access to view submissions of another form. MVP only requires a single form editor. How we allow multiple editors will depend on #1043"

**Comment by kevincrafts** [comment]
I suggest that we put the user reference field in a field set labeled Form Submissions

**Comment by owenmorrill** [comment]
Well, trying to parse the entity reference field to validate against role ids results in a massive pile of junk. Still can't find a way to filter users from the actual autocomplete field search bit itself.

`... (Array, 14 elements)
0 (Array, 2 elements)
target_id (String, 0 characters )
_weight (String, 1 characters ) 0
1 (String, 16 characters ) Add another item
2 (Array, 2 elements)
target_id (String, 10 characters ) crafts (2)
_weight (String, 1 characters ) 0
3 (Array, 2 elements)
target_id (String, 10 characters ) brokaw (9)
_weight (String, 1 characters ) 1
4 (Array, 2 elements)
target_id (String, 12 characters ) alfi2595 (6)
_weight (String, 1 characters ) 2
5 (Array, 2 elements)
target_id (String, 19 characters ) osr-test-owner (14)
_weight (String, 1 characters ) 3
6 (String, 16 characters ) Add another item
7 (String, 0 characters )
8 (String, 1 characters ) A
9 (String, 10 characters ) crafts (2)
10 (String, 10 characters ) brokaw (9)
11 (String, 12 characters ) alfi2595 (6)
12 (String, 19 characters ) osr-test-owner (14)
13 (String, 1 characters ) A G`

**Comment by owenmorrill** [comment]
https://github.com/CuBoulder/cu_forms_bundle/pull/2

**Comment by owenmorrill** [comment]
New PR uses radio buttons with a views entity reference display to show only users with the correct roles as options for submission viewers and form owners.

**Comment by owenmorrill** [comment]
Getting realname by using a simple db_query. https://github.com/CuBoulder/express/pull/2010/files#diff-10a036c9c6aaa56c2f31e853c8100fdaR82

**Comment by owenmorrill** [comment]
https://github.com/CuBoulder/express_homepage/blob/4059c42c2000d727b2809b24f835dcc25b644369/modules/custom/cu_directory/lib/cudirectory.class.php

Doesn't allow you to use an email string as the query. Returns empty.

`

function cu_ldap_page_alter(&$page) {
  $results = cu_directory_get_people($qstring = 'owen.morrill@colorado.edu', $scope = 'all');
  dpm($results);
}`

![invite-alter](https://user-images.githubusercontent.com/5008565/37614281-f49e3df0-2b6f-11e8-9f05-db83f001745a.png)

**Comment by owenmorrill** [comment]
https://github.com/CuBoulder/cu_forms_bundle/pull/2
